matrix.md

```diff
@@ -0,0 +1,128 @@
+# Compartmentalization Evaluation Matrix
+
+> Complete this matrix for a single implementation or system.  
+> Use ✅ / ❌ for checkboxes, categorical scales, and short text values as indicated.
+
+---
+
+## Security
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| **Isolation Model** | | |
+| Primary use case | Untrusted component isolation / Secret protection / Fault isolation / Other: ___ | |
+| Subject selection | Code-centric / Data-centric / Hybrid | |
+| Code-centric granularity | Function / Library / Thread / Process / VM / N/A | |
+| Data-centric granularity | Object / Region / Page / Allocation / File / N/A | |
+| **Isolation Approach** | | |
+| Hardware primitive(s) | MPK or PKU / MTE / CHERI / TEEs / Other: ___ / None | |
+| Software technique(s) | SFI / Boundary wrappers/marshalling / Memory-safe language / Language runtime / Other: ___ / None | |
+| Isolation abstraction | Process / Thread / Intra-Address Space Domain / VM / Container / Other: ___ | |
+| Requires runtime software support | ✅ / ❌ (if yes, describe: ___) | |
+| **Properties Enforced** | | |
+| Security properties provided | [Describe: e.g., "Memory integrity and confidentiality, syscall filtering"] | |
+| **Resilience** | | |
+| Compartment crashes isolated | ✅ / ❌ | |
+| **Interface Safety** | | |
+| Provides means to facilitate boundary checking/validation | ✅ / ❌ / Partial | |
+| **Trust & Threats** | | |
+| TCB includes | Compiler / OS / Firmware / CPU / Other: ___ (list applicable) | |
+| TCB approximate size | Small / Medium / Large / Very Large or [LOC/component count if known] | |
+| Side-channel resistance | ✅ / ❌ (if yes, describe mitigations: ___) | |
```

Comment:
Are we restricting this to cache-based side channels, or others? We might want to be more specific here, or at least give specific examples.